packages/astro/src/@types/astro.ts

```diff
@@ -2408,6 +2408,133 @@ export interface AstroUserConfig {
 		 * For a complete overview and the full API reference, see [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md) and [share your feedback](https://github.com/withastro/roadmap/pull/982).
 		 */
 		contentLayer?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.tsconfig
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.15.0
+		 * @description
+		 *
+		 * Enables the generation of `.astro/tsconfig.json`. This allows to exclude the `outDir` by default to avoid `astro check` false positives.
+		 *
+		 * If you enable this option, you'll need to update your `tsconfig.json` with a new `extends` value:
+		 *
+		 * ```json title="tsconfig.json" del={2} ins={3}
+		 * {
+		 *   extends: 'astro/tsconfigs/base',
+		 * 	 extends: ['astro/tsconfigs/base', './.astro/tsconfig.json']
+		 * }
+		 * ```
+		 *
+		 * If you have `include` or `exclude` inside your root `tsconfig.json`, you'll need to move them to your Astro config:
+		 *
+		 * ```json title="tsconfig.json" del={2-3}
+		 * {
+		 *   include: ['foo'],
+		 *   exclude: ['bar']
+		 * }
+		 * ```
+		 *
+		 * ```js title="astro.config.*" ins={6-7}
+		 * import { defineConfig } from 'astro/config'
+		 *
+		 * export default defineConfig({
+		 *   experimental: {
+		 *    tsconfig: {
+		 *      include: ['foo'],
+		 *      exclude: ['bar']
+		 *    }
+		 *   }
+		 * })
+		 * ```
+		 */
+		tsconfig?: {
+			/**
+			 * @docs
+			 * @name experimental.tsconfig.include
+			 * @type {Array<string>}
```

Comment:
```suggestion
			 * kind: h4
			 * @type {Array<string>}
```
These are all missing the heading level (I know, I still have to add those docs to Astro Docs Docs!)

Headings are rendered as `h2`, regular entries are rendered as `h3`, to have sub properties, you'll need to add the heading level manually.